src/jltypes.c

```diff
@@ -881,29 +881,57 @@ static jl_value_t *inst_datatype(jl_datatype_t *dt, jl_svec_t *p, jl_value_t **i
     if (istuple && ntp > 0 && jl_is_vararg_type(last)) {
         // normalize Tuple{..., Vararg{Int, 3}} to Tuple{..., Int, Int, Int}
         jl_value_t *va = jl_unwrap_unionall(last);
+        jl_value_t *va0 = jl_tparam0(va), *va1 = jl_tparam1(va);
         // return same `Tuple` object for types equal to it
-        if (ntp == 1 && jl_tparam0(va) == (jl_value_t*)jl_any_type &&
-            jl_is_unionall(last) && jl_tparam1(va) == (jl_value_t*)((jl_unionall_t*)last)->var) {
+        if (ntp == 1 &&
+            (last == (jl_value_t*)jl_vararg_type ||  // Tuple{Vararg} == Tuple
```

Comment:
I feel fairly certain you need to change the subtyping rules first, otherwise I think you'll run into trouble with all of the ways this can be varied (for example, `Tuple{Any, Vararg}`, `Tuple{Any, Vararg{T} where T)`, etc) and having construction normalization disagree with the current subtyping rules (which currently normalize in the other direction).